### PR TITLE
PR from upstream development

### PR DESCRIPTION
**Pull request by edwardchalstrey1**
This should be merged and tested in advance of #91 

TODO:
- [x] Resolve conflicts
    - ~~Make a new branch based on the Qing-dev merge on the main repo~~ - don't need to do this now @MajidBenam has fixed conflicts
- [x] Try out the updated dev branch locally and on seshatdata.com and get Majid to review
    - [x] Make a note of any additional steps that need to be done e.g. migrations (in fact, I will need to check there aren't any migrations with overlapping numbers. I might make a second branch to resolve this after this PR is merged)

### CODE REVIEW

**Comment by MajidBenam** [comment]
Hi @edwardchalstrey1 , I have fixed some minor conflicts, but I think there is not much more. I will let you do the merge after all tests pass.

**Comment by edwardchalstrey1** [comment]
> Hi @edwardchalstrey1 , I have fixed some minor conflicts, but I think there is not much more. I will let you do the merge after all tests pass.

Thanks @MajidBenam - actually don't worry about the tests passing, the current failure is unrelated to the code and due to the tests not being able to run when making a PR from a fork. I aim to fix this, but it already won't be an issue when we make PRs within this repo.

For now, I will leave this PR open. ~~Any further commits you make to Qing branch on your repo will show up here.~~ **Edit:** actually I noticed you already switched it to a new `majid-qing-merge` branch nice!

We can save hitting the merge button until right before you are ready to do #91 
Or let me know if you are ready for that already.

**Comment by edwardchalstrey1** [comment]
Ok @MajidBenam this did introduce a bug, which I can look into later - #187 

If we try to only make smaller pull requests going forward and work on this repo I think this can be avoided in future.

File: seshat/apps/sc/models.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 ########## Beginning of tuple choices for general Models
 ABSENT_PRESENT_CHOICES = (
 ('present', 'present'),
+('uncoded', 'uncoded'),
```
Comment (edwardchalstrey1) [inline seshat/apps/sc/models.py:26]: This addition is what resulted in this bug: #187

**Pull request merged**